batch7crud-roll3/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,181 @@
+			throws ServletException, IOException {
+		String pathInfo = request.getPathInfo();
+		String[] pathParts = pathInfo.split("/");
+		
+		try {
+			if (pathParts[1].equals("create")) {
+				createProcess(request, response);
+			} else if (pathParts[2].equals("edit")) {
+				editProcess(request, response, Integer.parseInt(pathParts[1]));
+			} else if (pathParts[2].equals("delete")) {
+				deleteProcess(request, response, Integer.parseInt(pathParts[1]));
+			} else {
+				list(request, response, 1);
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	private void list(HttpServletRequest request, HttpServletResponse response, int page)
+			throws ServletException, IOException {
+		try {
+			int pageSize = 3;
+			StudentService stdService = new StudentService();
+			List<Student> stdList = stdService.fetch(page, pageSize);
+			int count = stdService.fetchTotal();
+			
+			request.setAttribute("stdList", stdList);
+			request.setAttribute("pageSize", pageSize);
+
+			request.setAttribute("count", count);
+			request.setAttribute("pageNum", page);
+			request.getRequestDispatcher("/WEB-INF/views/student/list.jsp").forward(request, response);
+		} catch (DataException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			request.setAttribute("message", e.getMessage());
+			request.getRequestDispatcher("/WEB-INF/views/error.jsp").forward(request, response);
+		}
+	}
+
+	private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		request.getRequestDispatcher("/WEB-INF/views/student/create.jsp").forward(request, response);
+	}
+
+	private void createProcess(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+		String roll = request.getParameter("roll");
+		String name = request.getParameter("name");
+
+		Student student = new Student();
+		student.setRoll(Integer.parseInt(roll));
+		student.setName(name);
+
+		StudentService stdService = new StudentService();
+		try {
+			stdService.insert(student);
+			response.sendRedirect("/batch7crud-roll3/student");
+		} catch (DataException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			request.setAttribute("message", e.getMessage());
+			response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+			request.getRequestDispatcher("/WEB-INF/views/error.jsp").forward(request, response);
+		}
+
+	}
+
+	private void edit(HttpServletRequest request, HttpServletResponse response, int id)
+			throws ServletException, IOException {
+		StudentService stdService = new StudentService();
+		Student student;
+		try {
+			student = stdService.fetchStudentById(id);
+			request.setAttribute("student", student);
+			request.getRequestDispatcher("/WEB-INF/views/student/edit.jsp").forward(request, response);
+		} catch (DataException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)